server/match_handler.go

```diff
@@ -142,9 +143,10 @@ func NewMatchHandler(logger *zap.Logger, config Config, sessionRegistry SessionR
 		PresenceList:   presenceList,
 		core:           core,
```

Comment:
Let's also flip the `core` field to be exported as `Core` here, it will be needed later.